go/libraries/doltcore/merge/violations_fk.go

```diff
@@ -369,17 +369,30 @@ func parentFkConstraintViolations(
 func childFkConstraintViolations(
 	ctx context.Context,
 	foreignKey doltdb.ForeignKey,
-	postParent, postChild *constraintViolationsLoadedTable,
-	preChildSch schema.Schema,
+	postParent, postChild, preChild *constraintViolationsLoadedTable,
 	preChildRowData durable.Index,
-	receiver FKViolationReceiver) error {
+	receiver FKViolationReceiver,
+) error {
 	if preChildRowData.Format() == types.Format_DOLT {
+		if empty, err := preChildRowData.Empty(); err != nil {
+			return err
+		} else if empty {
+			emptyIdx, err := durable.NewEmptyIndex(ctx, postChild.Table.ValueReadWriter(), postChild.Table.NodeStore(), postChild.Schema)
```

Comment:
it'd be a bit nicer organizationally if it is possible to condense the empty setting in the above call next to the empty primary index check , something like preChild.IndexData = NewEmptyIndex. then you could just go into the secondary diff here